pkg/controller/endpointslice/metrics/cache.go

```diff
@@ -89,7 +89,13 @@ func (spc *ServicePortCache) totals(maxEndpointsPerSlice int) (int, int, int) {
 	for _, eInfo := range spc.items {
 		endpoints += eInfo.Endpoints
 		actualSlices += eInfo.Slices
-		desiredSlices += numDesiredSlices(eInfo.Endpoints, maxEndpointsPerSlice)
+		if eInfo.Endpoints > 0 {
+			desiredSlices += numDesiredSlices(eInfo.Endpoints, maxEndpointsPerSlice)
```

Comment:
Also, do we need to change endpointslicemirroring at all?

Reply:
yeah, I think that is a matter of where we want to special-case the case "there is always one slice, despite there are not endpoints at all" ... I took this approach because `numDesiredSlices` is also used in

https://github.com/aojea/kubernetes/blob/ee08c977fd2e44ceb10506290fbaaf6b242d6a99/pkg/controller/endpointslicemirroring/metrics/cache.go#L139

Reply:
hmm, seems you are right

Reply:
th eendpointslicemirroring seems to not use a placeholder, and already returns 0 slices for 0 endpoints, but better @robscott for confirming